Update DeferredCsrfToken to implement Supplier

kwondh5217 opened the pull request:
Closes gh-16870

<!--
For Security Vulnerabilities, please use https://pivotal.io/security#reporting
-->

<!--
Before creating new features, we recommend creating an issue to discuss the feature. This ensures that everyone is on the same page before extensive work is done.

Thanks for contributing to Spring Security. Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with gh-).
-->

jzheaux commented:
Hi, @kwondh5217! Thanks for the PR. Will you please update the commit to include `Closes gh-16870`? This helps when folks are researching the codebase to learn why a change was made. Something like this:

```
DeferredCsrfToken Implements Supplier

Closes gh-16870
```

jzheaux commented:
Also, @kwondh5217, will you please update any `2002-20xx` copyright years to be `2002-2025`?

kwondh5217 commented:
Hi @jzheaux, I’ve updated the commit message to include Closes gh-16870 and fixed the copyright years to 2002-2025.

Let me know if there's anything else to improve. Thanks for the review!

jzheaux commented:
Thanks for the PR, @kwondh5217! This has been merged into main as 9908d966448ca7416887ad44058913becece331c and will go out in the next release.

It was a different commit because for an unknown reason, the CI build didn't kick off to confirm the build passed, so I needed to verify that locally. 

In the future, if you submit a PR using a branch instead of using `main`, I can push these updates to your branch and then merge in GitHub. This will make the PR say "Merged" instead of "Closed".

kwondh5217 commented:
@jzheaux Thanks! Really appreciate your help and the tip!
